packages/server/modules/core/domain/objects/types.ts

```diff
@@ -20,9 +21,9 @@ export type InsertableSpeckleObject = NullableKeysToOptional<
 >
 
 export type RawSpeckleObject = Record<string, unknown> & {
-  id: string
-  speckle_type: string
-  totalChildrenCount: number
+  id?: string
```

Comment:
this is infact the case of the insertion object, but i do not want to encourage this.

we're so heavily relying on the `id` and the `speckle_type` and `__closure` property, that it doesn't make sense to allow those to be undefined in our type system.

Reply:
@gjedlicska I don't want to go into refactoring mode here, I'm only trying to describe how it works right now. The fact is that through the GQL API and possibly even the REST API you can submit any kind of arbitrary object and we submit all kinds of junk objects through tests.

If we want to change this after, I'm open to it, but not within these multi region PRs. If you want I can leave a comment next to the types outlining your concerns